### DISCUSSION TITLE
VCC value command on precompiled version

### DISCUSSION THREAD

**Discussion by 18923** [arendst/Tasmota#15288] [Q&A] [answered]
I have an ESP8266 in a mailbox down the driveway running off an 18650 battery with regulator. When mail arrives power is applied to the ESP, and a "Mail Arrived" and "VCC Level" message is sent to my Google home. I used the VCC level as an indicator when to replace/recharge the battery.
I recently had to replace the ESP8266 with a new one loaded with the latest precompiled version 11, I think the old one was 6.1 or so.
Now I no longer get the VCC level, as per release notes VCC level is no longer provided by default. Is there a command to enable VCC on the precompiled version or do I have to compile my own to get the VCC value function?

**Comment by barbudor** [comment, marked as answer]
Indeed, the ADC_VCC mode is not enabled by default and you need a self compiled version
Details are there : https://tasmota.github.io/docs/ADC/#adc_vcc

However as you are using a LiIon battery with a regulator, I would better advice to connect the battery level to the ADC input pin through a voltage divider to give you an exact measurment of the battery level.
As you were measuring the VCC level, after the regulator, you would see a constant voltage until the battery is lower than 3V3 + DropOutVoltage which means that from that point, voltage may keep on dropping very fast.
On my solar+liion powered sensor, I'm using that to have an exact measurment of the battery voltage from 4.2V
I'm using a low drop out regulator at 3.0V which allow me to use the battery down to 3.2V safely.
After that, LiIon drops very quickly.

Nota that LiIon may not be the best technology for outdoor devices as they may loose capacity in cold time. Alkaline are better "all weather" capable

**Reply by 18923** [reply]
Thank you for the feedback, my original ESP8266 did not have an analog input. Although the new one has ADC (D1 mini) I could use the method you suggested but the old system has worked well for me, been running about 4 years. I recharge the battery about once a month. 
Using Node Red I really only get notified when VCC is below 3.2v.
As a quick fix I may go back to Tasmota Version 6.x.
BTW I live in Canada and the 18650 battery has worked well all winter. I like your idea of a solar cell so I may do that in the future.

**Reply by barbudor** [reply]
You can easily recompile a Tasmota version with ADC_VCC using TasmoCompiler for example

**Reply by sfromis** [reply]
If you can make do with `tasmota-lite`, that binary [seems](https://tasmota.github.io/docs/Firmware-Builds/) to have `USE_ADC_VCC` included by default.

**Reply by barbudor** [reply]
If you want to switch to measuring the full battery voltage note that the mini-D1 already has a voltage divider made from 2 resistors where A0 is the pin of the mini-D1 PCB and ADC is the ADC pin of the ESP8266

![image](https://user-images.githubusercontent.com/2996042/161579732-99f01e55-16ec-4ee2-8ce2-26e93dddc52f.png)

As the ESP's  can support a max of 1.0V, with this 100k/(100k+220k) divider, the mini-D1 can already measure up to 3.2V
If you want to measure the LiIon, you just need to add another 220K in between the A0 pin and the LiIon (+). THat will gilve you a full range of 4.4V.
Then using AdcParam you can remap the ADC 0..1024 range to 0..4400 to get the value in milliVolts (AdcRange works on integers)
`AdcParam 6, 0, 1023, 0, 4400`

**Reply by 18923** [reply]
Great information, I will look into TasmoCompiler  and tasmota-lite.
79 times around the Sun and still learning.

**Reply by sfromis** [reply]
For battery builds, it could also be a consideration to use a LiFePO4 cell, working voltage ~3.0-3.2V (max 3.65V).

**Reply by 18923** [reply]
"tasmota-lite" worked great, it will be my goto version from now on.

Thank you
